src/ape/api/projects.py

```diff
@@ -0,0 +1,50 @@
+from typing import TYPE_CHECKING, Optional
+
+from ethpm_types import PackageManifest
+
+from ape.types import DependencyType
+from ape.utils import abstractdataclass, abstractmethod
+
+if TYPE_CHECKING:
+    from ape.managers.compilers import CompilerManager
+
+
+@abstractdataclass
+class DependencyAPI:
```

Comment:
This is not the `ProjectAPI` we talked about - this refers to like Github, Gitlab, etc.
The Github implementation is found in `ape.managers.project` currently.